doc/dev.md

```diff
@@ -0,0 +1,50 @@
+# Local development
+
+This document sets you up to use locally running instances of `ocaml-ci-server` and `ocaml-ci-web` to build an OCaml project that is in a repository owned by your GitHub user.
+
+### Setting up a GitHub App
+
+Since `ocaml-ci` is a GitHub App, create a GitHub App under your own user and point it to localhost via a webhook payload delivery service like [smee.io](https://smee.io).
+
+To do this, follow the instructions in [Setting up your development environment to create a GitHub App](https://docs.github.com/en/developers/apps/getting-started-with-apps/setting-up-your-development-environment-to-create-a-github-app) but when it comes to setting permissions for your app, set the following:
+
+```
+Metadata: Read
+Checks: Read and write
+Commit statuses: Read and write
```

Comment:
Yes it should plus `Webhooks - Read only`

You'll also want to subscribe to the events Pull Request and Check Run.

Reply:
Ok thanks - I think these relate to the additional permissions that are required for PRs and since I have not used the app with any PRs, I haven't run across them.